File: src/examples/cpp_pubsub/src/publisher.cpp
#include <chrono>    // chrono_literals
#include <functional>// bind
#include <memory>    // make_shared
#include <string>    // to_string

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

using namespace std::chrono_literals;

class Publisher : public rclcpp::Node {
public:
    Publisher() : Node("publisher") {
        publisher_ = this->create_publisher<std_msgs::msg::String>("topic", 10);
        timer_ = this->create_wall_timer(500ms, std::bind(&Publisher::timer_callback, this));
        count_ = 0;
        RCLCPP_INFO(this->get_logger(), "Publisher has been started");
    }

private:
    void timer_callback() {
        auto msg = std_msgs::msg::String();
        msg.data = "Hello World number " + std::to_string(count_++);
        publisher_->publish(msg);
        RCLCPP_INFO(this->get_logger(), ("Message has been published " + msg.data).c_str());
    }

    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
    size_t count_;
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto publisher = std::make_shared<Publisher>();
    rclcpp::spin(publisher);
    rclcpp::shutdown();
    return 0;
}

File: src/examples/py_pubsub/py_pubsub/publisher.py
#!/usr/bin/env python3
import typing as t

import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class Publisher(Node):  # type: ignore
    def __init__(self) -> None:
        super().__init__("publisher")
        self.publisher_ = self.create_publisher(String, "topic", 10)
        self.timer_ = self.create_timer(0.5, self.timer_callback)
        self.count_ = 0
        self.get_logger().info("Publisher has been started")

    def timer_callback(self) -> None:
        msg = String()
        msg.data = f"Hello World number {self.count_}"
        self.publisher_.publish(msg)
        self.count_ += 1
        self.get_logger().info(f"Message has been published {msg.data}")


def main(arguments: t.List[str] | None = None) -> None:
    rclpy.init(args=arguments)
    publisher = Publisher()
    rclpy.spin(publisher)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
